march_joint_inertia_controller/include/march_joint_inertia_controller/joint_inertia_controller.h

```diff
@@ -3,23 +3,82 @@
 #ifndef MARCH_HARDWARE_JOINT_INERTIA_CONTROLLER_H
 #define MARCH_HARDWARE_JOINT_INERTIA_CONTROLLER_H
 
+#include <control_toolbox/pid.h>
 #include <controller_interface/controller.h>
-#include <hardware_interface/joint_command_interface.h>
-#include <pluginlib/class_list_macros.h>
+#include "march_joint_inertia_controller/inertia_estimator.h"
+#include <std_msgs/Float64.h>
 
 namespace joint_inertia_controller
 {
-class InertiaController : public controller_interface::Controller<hardware_interface::PositionJointInterface>
+class InertiaController : public controller_interface::Controller<hardware_interface::EffortJointInterface>
 {
 public:
+  struct Commands
+  {
+    double position_;    // Last commanded position
+    double velocity_;    // Last commanded velocity
+    bool has_velocity_;  // false if no velocity command has been specified
+  };
+
+  InertiaController();
+  ~InertiaController();
+
   bool init(hardware_interface::PositionJointInterface* hw, ros::NodeHandle& n);
   void update(const ros::Time& time, const ros::Duration& period);
   void starting(const ros::Time& time);
   void stopping(const ros::Time& time);
```

Comment:
I know this was already merged, but it is still important
```suggestion
  bool init(hardware_interface::PositionJointInterface* hw, ros::NodeHandle& n) override;
  void update(const ros::Time& time, const ros::Duration& period) override;
  void starting(const ros::Time& time) override;
  void stopping(const ros::Time& time) override;
```

Reply:
I just tested this, but the compiler seems to think it doesn't override anything

Reply:
That is because this `init` still accepts the `PositionJointInterface` and not the `EffortJointInterface`.